server/dao/RideDAO.php

```diff
@@ -35,7 +43,14 @@ public function getRides($page = 0, $numberPerPage = 10)
             $stmt->execute();
             $stmt->setFetchMode(PDO::FETCH_CLASS, "Ride");
             while ($ride = $stmt->fetch()) {
-                $data[] = $ride->getRideInfo();
+                $ride = $ride->getRideInfo();
+
+                if ($populate)
+                {
+                   $this->populateIDs($ride);
```

Comment:
Will work for now but we will probably want to switch this to a join for production.